### PR TITLE
[FIX] Patches multiple streamlit bugs

### LINKED ISSUE #184: streamlit_app can't find model_db
After bringing up the backend with the streamlit front end, Streamlit can't find model_db. 

It looks like utils.py is missing some bits that should point it to core-api rather than its own models.

**To Reproduce**

```console
docker compose up core-api ingester embedder streamlit_app
```

Go to `localhost:8501`.

```text
2024-04-05 08:35:11 
2024-04-05 08:35:11 Collecting usage statistics. To deactivate, set browser.gatherUsageStats to False.
2024-04-05 08:35:11 
2024-04-05 08:35:13 
2024-04-05 08:35:13   You can now view your Streamlit app in your browser.
2024-04-05 08:35:13 
2024-04-05 08:35:13   URL: http://0.0.0.0:8501
2024-04-05 08:35:13 
2024-04-05 08:37:26 2024-04-05 07:37:26.035 Uncaught app exception
2024-04-05 08:37:26 Traceback (most recent call last):
2024-04-05 08:37:26   File "/app/.venv/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 542, in _run_script
2024-04-05 08:37:26     exec(code, module.__dict__)
2024-04-05 08:37:26   File "/app/Welcome.py", line 2, in <module>
2024-04-05 08:37:26     from utils import init_session_state
2024-04-05 08:37:26   File "/app/utils.py", line 28, in <module>
2024-04-05 08:37:26     from model_db import SentenceTransformerDB
2024-04-05 08:37:26 ModuleNotFoundError: No module named 'model_db'
```

**Expected behaviour**

Streamlit should use the core-api.

### LINKED ISSUE #203: Elasticsearch retriever in streamlip-app doesn't work without subscription
[RRF (`ApproxRetrievalStrategy(hybrid=True)`) requires Platinum or Enterprise to work.](https://www.elastic.co/subscriptions) and retrieval breaks without it.

Suggest setting hybrid to `False` as default and potentially control with an `ELASTIC_SUBSCRIPTION_LEVEL` environment variable (one of `basic`, `platinum`, `enterprise`)?

### PR DESCRIPTION
**Pull request by wpfl-dbt**
## Context

The streamlit app wasn't useable. Lots of core features simply don't work with the Elasticsearch backend. This fixes as many bugs as I could find.

While #185 closes #184 , this _definitely_ super-closes it.

## Changes proposed in this pull request

* Allows use of basic Elasticsearch tier, turning off hybrid by default (closes #203 )
    * Adds `ELASTIC_SUBSCRIPTION_LEVEL` environment variable
* Mounts streamlit code for quicker dev loop
* Parses string UUIDs returned from Elasticsearch and turns them into UUIDs: Pydantic was throwing tonnes of errors
    * This results in a bit of datatype wrangling in the various Streamlit pages
* Changes the `change_selected_model()` function so it doesn't print an API key for all to see
* Refactors `get_files_by_uuid()` into utils to reduce redundancy
* Fixes multiple calls to `st.session_state.storage_handler` that used the old `item_uuid=file.uuid, model_type="File"` arguments rather than the `PersistableModel` object directly
* Fixes issue where streamlit was making new buckets instead of reading from the environment
* Ran pre-commit on `streamlit_app/` and `redbox/`

## Guidance to review

I don't know how to run integration tests and would appreciate help if that needs doing.

* `docker compose down`
* `rm -rf data/*`
* `docker compose up core-api embedder ingester streamlit-app --build`
* Play with the app, enjoy it working

## Relevant links

* [Inspiration for the `AfterValidator` code](https://github.com/pydantic/pydantic/discussions/7023)

## Things to check

- [x] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests
- [x] I have manually tested the streamlit app

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -45,6 +45,7 @@ services:
       - 8501:8501
     volumes:
       - ./data:/app/data:Z
+      - ./streamlit_app:/app/streamlit_app/
```
Comment (gecBurton) [inline docker-compose.yml:48]: :+1:

File: redbox/models/base.py [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+from typing import Annotated, Optional
 from uuid import UUID, uuid4
 
-from pydantic import BaseModel, Field, computed_field
+from pydantic import AfterValidator, BaseModel, Field, computed_field
 
 
 class PersistableModel(BaseModel):
-    uuid: UUID = Field(default_factory=uuid4)
+    uuid: UUID | Annotated[str, AfterValidator(lambda x: UUID(x, version=4))] = Field(
```
Comment (gecBurton) [inline redbox/models/base.py:9]: v nice, i like this approach, one minor suggestion:

```suggestion
    uuid: UUID | Annotated[str, AfterValidator(UUID)] = Field(
```
as the version doesnt apply in this case where we are just casting it?

Comment (lmwilkigov) [reply redbox/models/base.py:9]: I think the default factory is still useful here as the generator for when nothing is passed in. Both validation approaches look valid to me, shall leave it up to your taste!

**Review by lmwilkigov — APPROVED** [review]
This is fantastic stuff @wpfl-dbt! Much cleaner type handling across the board and great spot on the elastic

**Comment by gecBurton** [comment]
@wpfl-dbt you can either run integration tests via github actions (workflow dispatch) or via `make test-integration` but you dont need to do this as I dont think your changes will have any impact

**Review by gecBurton — APPROVED** [review]
massive thank you @wpfl-dbt !

**Pull request merged**